code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1279,21 +1315,32 @@ func (sc *StorageSmartContract) finalizedPassRates(alloc *StorageAllocation) ([]
 	for _, ba := range alloc.BlobberAllocs {
 		if ba.Stats == nil {
 			ba.Stats = new(StorageAllocationStats)
+			passRates = append(passRates, 1.0)
+			continue
 		}
 		ba.Stats.FailedChallenges += ba.Stats.OpenChallenges
 		ba.Stats.OpenChallenges = 0
-		ba.Stats.TotalChallenges = ba.Stats.FailedChallenges + ba.Stats.SuccessChallenges
-		if ba.Stats.TotalChallenges == 0 {
+
+		baTotal := ba.Stats.FailedChallenges + ba.Stats.SuccessChallenges
```

Comment:
should we update ``ba.Stats.TotalChallenges`` with this value?

Reply:
No, the `ba.Stats.TotalChallenges` will only be updated when new challenges are generated, update here could lead to wrong stats.